iam-login-service/src/main/java/it/infn/mw/iam/api/client/registration/service/DefaultClientRegistrationService.java

```diff
@@ -345,7 +346,12 @@ public RegisteredClientDTO registerClient(RegisteredClientDTO request,
     checkAllowedGrantTypes(request, authentication);
     cleanupRequestedScopes(client, authentication);
 
+    if (isNull(client.getClientSecret())) {
+      client.setClientSecret(defaultsService.generateClientSecret());
+    }
+    String tmpClientSecret = client.getClientSecret();
     client = clientService.saveNewClient(client);
+    client.setClientSecret(tmpClientSecret);
```

Comment:
Non mi è chiaro perché stiamo chiamando `getClientSecret` e `setClientSecret` due volte. Te lo puoi salvare da qualche parte, fare la verifica se è null e nel caso lo sovrascrivi

```java
String secret = client.getClientSecret();
if (isNull(maybeSecret)) {
  secret = defaultsService.generateClientSecret();
  client.setClientSecret(secret);
}
client = clientService.saveNewClient(client);
```

che in realtà ancora più semplicemente diventa

```java
if (isNull(client.getClientSecret())) {
  client.setClientSecret(defaultsService.generateClientSecret());
}
client = clientService.saveNewClient(client);
```

non è sufficiente una cosa del genere?

Reply:
Ma soprattutto, perché facciamo tutta sta roba? Non dobbiamo toccare questo service, è sufficiente fare l'encoding al momento del salvataggio del client `saveNewClient` (vedi sopra).
Se non è necessario buttiamo via le modifiche a questo file